pkg/kubectl/cmd/annotate.go

```diff
@@ -107,7 +109,7 @@ func NewCmdAnnotate(f cmdutil.Factory, out io.Writer) *cobra.Command {
 
 	cmd := &cobra.Command{
 		Use:     "annotate [--overwrite] (-f FILENAME | TYPE NAME) KEY_1=VAL_1 ... KEY_N=VAL_N [--resource-version=version]",
-		Short:   "Update the annotations on a resource",
+		Short:   T("annotate_short"),
```

Comment:
> Don't we need stable keys to be able to locate the translated string? 

A key only have to be "stable" while its message doesn't change. If the message changes it has to be translated again, right? So it becomes a new message anyway, with a new key. 

That said, you generate the keys, as in 

```
T("Update the annotations on a resource")
```

becomes for example the key

```
kubectl-annotate#update-the-annotations-on-a-resource
```

`T` takes a message string, generate the key from it and search for that key in the translated file for the language you want. If not found, just use the default message, which is just `T` argument.

> It's also not obvious how this approach would deal with, "I want the same message as that guy over there".

I don't recommend reusing messages, even if they are as simple as a single word. Every single call to `T` must relate directly to one dedicated message key. Exception being maybe inside the same command. But duplicated strings in translation files are ok and expected.